src/Microsoft.AspNetCore.Mvc.Core/Filters/FilterCollection.cs

```diff
@@ -83,6 +129,22 @@ public IFilterMetadata AddService(Type filterType)
         /// <summary>
         /// Adds a type representing an <see cref="IFilterMetadata"/>.
         /// </summary>
+        /// <typeparam name="TFilterType">Type representing an <see cref="IFilterMetadata"/>.</typeparam>
+        /// <param name="order">The order of the added filter.</param>
+        /// <returns>An <see cref="IFilterMetadata"/> representing the added service type.</returns>
+        /// <remarks>
+        /// Filter instances will be created through dependency injection. Use
+        /// <see cref="Add(Type)"/> to register a service that will be created via
+        /// type activation.
+        /// </remarks>
+        public IFilterMetadata AddService<TFilterType>(int order)
```

Comment:
you can add a `where TFilterType : IFilterMetadata` here, and then that validation will also happen at compile time